### PR TITLE
Add real build method

### PR DESCRIPTION
**Pull request by kevinrizza**
-Create build_bundle method
-Update identify package to better handle errors
-Add build method tests
-Add yaml and test packages to pipfile

### CODE REVIEW

File: command/build.py [hunk 1]
```diff
@@ -1,8 +1,39 @@
+import yaml as pyYaml
+import command.identify as identify
+
 class BuildCmd():
     name = 'build'
-
-    def __init__(self):	
+
```
Comment (SamiSousa) [inline command/build.py]: Remove unnecessary whitespace

File: command/build.py [hunk 2]
```diff
@@ -1,8 +1,39 @@
...
+            )
+        )
+        return operatorArtifactsTemplate
     
-    def build(self):
-        print("Hello, world. The build feature is currently under construction.")
+    def _get_field_entry(self, yamlContent):
+        yaml_type = identify.get_operator_artifact_type(yamlContent)
+        if yaml_type == "ClusterServiceVersion" or yaml_type == "CustomResourceDefinition" or yaml_type == "Package":
+            return yaml_type[0:1].lower() + yaml_type[1:] + 's'
```
Comment (SamiSousa) [inline command/build.py]: Do we need the if statement? The way this is used in `_updateArtifact` requires an output, so in the case that `yaml_type` is an unexpected value the program errors out; is this a "pythonic" way of erroring out? Should there be an error log as well?

Comment (awgreene) [reply command/build.py]: Actually the identify.get_operator_artifact_type method will raise an exception if the file isn't identified as a CSV, CRD, or Pacakge YAML - your points that the `if` statement isn't needed and logging are still valid.

File: command/build.py [hunk 3]
```diff
@@ -1,8 +1,39 @@
...
+
+    def _updateArtifact(self, operatorArtifact, yamlContent):
```
Comment (awgreene) [inline command/build.py]: I assume the underscore means this is a private method, in these cases do we not add method descriptions?

Comment (awgreene) [inline command/build.py]: can we change operatorArtifact to Bundle? Or operatorBundle?

Comment (awgreene) [inline command/build.py]: Can we change the name of this method from _updateArtifact to  _updateBundle?

File: command/build.py (hunk 1 above)
Comment (kevinrizza) [reply command/build.py]: @SamiSousa What is unnecessary about this whitespace? It makes the pace between setting name and the init method more readable.

File: command/build.py (hunk 3 above)
Comment (kevinrizza) [reply command/build.py]: @awgreene yes that is the case.

File: command/build.py (hunk 1 above)
Comment (SamiSousa) [reply command/build.py]: The newline is fine, but it's followed by 4 spaces which are unnecessary.

File: command/build.py (hunk 2 above)
Comment (kevinrizza) [reply command/build.py]: I don't think it makes sense to add a log when the raised exception will give the same information. I did address the comment related to removing the if statement --- we will never hit it.

**Comment by kevinrizza** [comment]
Closed by accident.....

File: command/build.py (hunk 1 above)
Comment (kevinrizza) [reply command/build.py]: Addressed in latest commit.

File: command/build.py (hunk 2 above)
Comment (kevinrizza) [reply command/build.py]: Addressed in latest commit.

File: command/build.py (hunk 3 above)
Comment (kevinrizza) [reply command/build.py]: Addressed in latest commit.

Comment (kevinrizza) [reply command/build.py]: Addressed in latest commit.

File: command/build.py [hunk 4]
```diff
@@ -1,8 +1,38 @@
...
+
 class BuildCmd():
     name = 'build'
 
-    def __init__(self):	
+    def __init__(self):
         pass
+
+    def _get_empty_bundle(self):
+        operatorBundleTemplate = dict(
```
Comment (awgreene) [inline command/build.py]: Could we simply return the dict?

File: command/identify.py [hunk 5]
```diff
@@ -1,13 +1,15 @@
 import yaml
 
 def get_operator_artifact_type(operatorArtifactString):
+    """get_operator_artifact_type takes a yaml string and determines if it is one of the expected bundle types
```
Comment (awgreene) [inline command/identify.py]: Could you change this to mention that it checks for the expected CSV, CRD, or Package type?

File: command/build.py (hunk 4 above)
Comment (kevinrizza) [reply command/build.py]: True. Updated in latest commit.

File: command/identify.py (hunk 5 above)
Comment (kevinrizza) [reply command/identify.py]: Updated in latest commit.

File: command/build.py [hunk 6]
```diff
@@ -1,8 +1,37 @@
...
+                customResourceDefinitions = [],
+                clusterServiceVersions = [],
+                packages = [],
+            )
+        )
     
-    def build(self):
-        print("Hello, world. The build feature is currently under construction.")
+    def _get_field_entry(self, yamlContent):
+        yaml_type = identify.get_operator_artifact_type(yamlContent)
```
[the remark this answers is not in the document]
Comment (aravindhp) [reply command/build.py]: Given we are going to be using `operator-registry`, we should be adopting [definitions](https://github.com/operator-framework/operator-registry#manifest-format) that are being used there.

[the remark this answers is not in the document]
Comment (kevinrizza) [reply command/build.py]: @SamiSousa Artifact in this context (identify.py) does not refer to the bundle, it refers to any subsection of yaml that is being created. Since we don't have a term for a generic 'component' I think `Artifact` is fine in this context.

File: command/build.py [hunk 7]
```diff
@@ -1,8 +1,37 @@
...
+        return yaml_type[0:1].lower() + yaml_type[1:] + 's'
+
+    def _updateBundle(self, operatorBundle, yamlContent):
+        operatorBundle["data"][self._get_field_entry(yamlContent)].append(pyYaml.load(yamlContent))
```
Comment (MartinBasti) [inline command/build.py]: Please always use `pyYaml.safe_load` to avoid security issues, unless you really need `yaml.load` and you can guarantee that `yamlContent` is safe.

File: command/build.py [hunk 8]
```diff
@@ -1,8 +1,37 @@
+import yaml as pyYaml
+import command.identify as identify
+
 class BuildCmd():
     name = 'build'
 
-    def __init__(self):	
+    def __init__(self):
```
Comment (MartinBasti) [inline command/build.py]: empty `__init__` can be omitted

File: command/identify.py [hunk 9]
```diff
@@ -1,13 +1,16 @@
...
         if "packageName" in operatorArtifact:
-            return "packages"
+            return "Package"
         elif "kind" in operatorArtifact:
-            if operatorArtifact["kind"] == "ClusterServiceVersion": 
-                return "clusterServiceVersions"
-            elif operatorArtifact["kind"] == "CustomResourceDefinition":
-                return "customResourceDefinitions"
-    return "invalid"
+            if operatorArtifact["kind"] == "ClusterServiceVersion" or operatorArtifact["kind"] == "CustomResourceDefinition":
```
Comment (MartinBasti) [inline command/identify.py]: just protip `if operatorArtifact["kind"] in ("ClusterServiceVersion", "CustomResourceDefinition"):`

File: tests/test_build.py [hunk 10]
```diff
@@ -0,0 +1,16 @@
+from unittest import TestCase
+import pytest
+from command.build import BuildCmd
+
+
+def test_create_bundle():
+    paths = ["tests/test_files/csv.yaml", "tests/test_files/crd.yaml", "tests/test_files/package.yaml"]
+    yamls = []
+    for x in range(3):
```
Comment (MartinBasti) [inline tests/test_build.py]:
```python
for path in paths:
    with open(path) as f:
        ...
```

File: command/build.py [hunk 11]
```diff
@@ -1,8 +1,37 @@
+import yaml as pyYaml
```
Comment (MartinBasti) [inline command/build.py]: in previous PR you used just `yaml` this is quite inconsistent naming

File: Pipfile.lock [hunk 12]
```diff
@@ -48,13 +62,65 @@
...
+            "version": "==1.7.0"
+        },
+        "pytest": {
+            "hashes": [
+                "sha256:41568ea7ecb4a68d7f63837cf65b92ce8d0105e43196ff2b26622995bb3dc4b2",
+                "sha256:c3c573a29d7c9547fb90217ece8a8843aa0c1328a797e200290dc3d0b4b823be"
+            ],
+            "version": "==4.1.1"
+        },
+        "pyyaml": {
```
Comment (MartinBasti) [inline Pipfile.lock]: Please use this yaml module, it is production stable, your version is in beta according pypi
https://pypi.org/project/PyYAML/

Comment (kevinrizza) [reply Pipfile.lock]: @MartinBasti What do you mean? The linked version (3.13) is listed as stable and is the version we are importing and using.

**Comment by kevinrizza** [comment]
@MartinBasti Addressed your comments in latest commit.

File: command/build.py (hunk 11 above)
Comment (kevinrizza) [reply command/build.py]: @MartinBasti addressed in latest commit.

File: tests/test_build.py (hunk 10 above)
Comment (kevinrizza) [reply tests/test_build.py]: @MartinBasti addressed in latest commit.

File: command/identify.py (hunk 9 above)
Comment (kevinrizza) [reply command/identify.py]: @MartinBasti addressed in latest commit.

File: command/build.py (hunk 8 above)
Comment (kevinrizza) [reply command/build.py]: @MartinBasti addressed in latest commit.

File: command/build.py (hunk 7 above)
Comment (kevinrizza) [reply command/build.py]: @MartinBasti addressed in latest commit.

File: Pipfile.lock (hunk 12 above)
Comment (MartinBasti) [reply Pipfile.lock]: Ah sorry, it was unexpected to see it in lowercase, I thought that this was https://pypi.org/project/pyaml/

So it is ok, sorry for the noise

File: command/build.py [hunk 13]
```diff
@@ -1,8 +1,32 @@
-class BuildCmd():
-    name = 'build'
+import yaml
+import command.identify as identify
```
Comment (MartinBasti) [inline command/build.py]: how about `from command import identify` ?

**Comment by MartinBasti** [comment]
LGTM

File: command/build.py [hunk 14]
```diff
@@ -1,8 +1,32 @@
...
+        )
     
-    def build(self):
-        print("Hello, world. The build feature is currently under construction.")
+    def _get_field_entry(self, yamlContent):
+        yaml_type = identify.get_operator_artifact_type(yamlContent)
+        return yaml_type[0:1].lower() + yaml_type[1:] + 's'
+
+    def _updateBundle(self, operatorBundle, yamlContent):
+        operatorBundle["data"][self._get_field_entry(yamlContent)].append(yaml.safe_load(yamlContent))
```
[the remark this answers is not in the document]
Comment (MartinBasti) [reply command/build.py]: Please see example: https://security.openstack.org/guidelines/dg_avoid-dangerous-input-parsing-libraries.html

[the remark this answers is not in the document]
Comment (awgreene) [reply command/build.py]: Thanks @MartinBasti !

File: command/build.py (hunk 13 above)
Comment (kevinrizza) [reply command/build.py]: addressed in latest commit
